src/Toolkit/tests/Functional/ComponentsRenderingTest.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Toolkit\Tests\Functional;
+
+use Spatie\Snapshots\Drivers\HtmlDriver;
+use Spatie\Snapshots\MatchesSnapshots;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\Filesystem\Path;
+use Symfony\Component\Finder\Finder;
+use Symfony\UX\Toolkit\Asset\Component;
+use Symfony\UX\Toolkit\Kit\Kit;
+use Symfony\UX\Toolkit\Kit\KitFactory;
+use Symfony\UX\Toolkit\Registry\LocalRegistry;
+
+class ComponentsRenderingTest extends WebTestCase
+{
+    use MatchesSnapshots;
+
+    private const KITS_DIR = __DIR__.'/../../kits';
+
+    /**
+     * @return iterable<string, string, string>
+     */
+    public static function provideTestComponentRendering(): iterable
+    {
+        foreach (LocalRegistry::getAvailableKitsName() as $kitName) {
+            $kitDir = Path::join(__DIR__, '../../kits', $kitName, 'docs/components');
+            $docsFinder = (new Finder())->files()->name('*.md')->in($kitDir)->depth(0);
+
+            foreach ($docsFinder as $docFile) {
+                $componentName = $docFile->getFilenameWithoutExtension();
+
+                $codeBlockMatchesResult = preg_match_all('/```twig.*?\n(?P<code>.+?)```/s', $docFile->getContents(), $codeBlockMatches);
+                if (false === $codeBlockMatchesResult || 0 === $codeBlockMatchesResult) {
+                    throw new \RuntimeException(\sprintf('No Twig code blocks found in file "%s"', $docFile->getRelativePathname()));
+                }
+
+                foreach ($codeBlockMatches['code'] as $i => $code) {
+                    yield \sprintf('Kit %s, component %s, code #%d', $kitName, $componentName, $i + 1) => [$kitName, $componentName, $code];
+                }
+            }
+        }
+    }
+
+    /**
+     * @dataProvider provideTestComponentRendering
+     */
+    public function testComponentRendering(string $kitName, string $componentName, string $code): void
+    {
+        $twig = self::getContainer()->get('twig');
+        $kitContextRunner = self::getContainer()->get('ux_toolkit.kit.kit_context_runner');
+
+        $kit = $this->instantiateKit($kitName);
+        $template = $twig->createTemplate($code);
+
+        try {
+            $renderedCode = $kitContextRunner->runForKit($kit, fn () => $template->render());
+        } catch (\Twig\Error\SyntaxError $e) {
+            if (\PHP_VERSION < 80200 && str_contains($e->getMessage(), 'Unknown "tailwind_merge" filter')) {
+                $this->markTestSkipped('Filter "tailwind_merge" is not supported on PHP <8.2.');
+            }
```

Comment:
And here, skip the test if the issue is caused by missing `tales-from-a-dev/twig-tailwind-extra` dependency, since we don't want to create "weird" snapshots.